File: ports/rust/src/gui/painting/bcspen.rs
use bcs_core::global::bcscoretypes::BcsGlobalColor;

// BcsPen defines how lines and outlines are drawn
pub struct BcsPen {
    pub color: BcsGlobalColor,
    pub width: i32,
}

impl BcsPen {
    pub fn new(color: BcsGlobalColor, width: i32) -> Self {
        BcsPen { color, width }
    }
}

// BcsBrush defines how shapes are filled
pub struct BcsBrush {
    pub color: BcsGlobalColor,
}

impl BcsBrush {
    pub fn new(color: BcsGlobalColor) -> Self {
        BcsBrush { color }
    }
}

File: ports/go/src/gui/painting/bcspen.go
package painting

import "bcs/core/global"

// BcsPen defines how lines and outlines are drawn
type BcsPen struct {
	color global.BcsGlobalColor
	width int
}

func NewBcsPen(color global.BcsGlobalColor, width int) *BcsPen {
	return &BcsPen{color: color, width: width}
}

// BcsBrush defines how shapes are filled
type BcsBrush struct {
	color global.BcsGlobalColor
}

func NewBcsBrush(color global.BcsGlobalColor) *BcsBrush {
	return &BcsBrush{color: color}
}
